core/src/main/scala/org/apache/spark/scheduler/BlacklistStrategy.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import scala.collection.mutable
+
+import org.apache.spark.Logging
+import org.apache.spark.SparkConf
+import org.apache.spark.util.SystemClock
+
+/**
+ * The interface to determine executor blacklist and node blacklist.
+ */
+trait BlacklistStrategy {
+  val expireTimeInMillisecond: Long
+
+  // TaskId is optional here because the BlacklistStrategy could be unrelated to TaskId
```

Comment:
You are right, it's not used now. But it might be necessary, to avoid allocating new YARN container on blacklisted node, "YarnSchedulerBackend" will call "nodeBlacklist" (nodeBlacklist could be depend on executorBlacklist, eg: if more than 3 blacked executor is on the same node, we think the node is also blacked), and YarnSchedulerBackend is not able to provide taskId.

And I agree that define two method instead is clearer and better interface design. I will remove Option here.